### PR TITLE
[GLT-3518] Slide count improvements

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/SampleStockDto.java [hunk 1]
```diff
@@ -10,8 +10,8 @@
 public class SampleStockDto extends SampleTissueDto {
 
   private Long tissueProcessingClassId;
-  private String concentration;
```
Comment (callunity) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/SampleStockDto.java:13]: Was `concentration` intended to be removed here?

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java [hunk 2]
```diff
@@ -331,20 +331,30 @@ private NamingScheme getNamingScheme(Sample sample) {
...
+      slidesConsumed = ((SampleTissuePiece) child).getSlidesConsumed();
+    } else if (isStockSample(child)) {
+      slidesConsumed = ((SampleStock) child).getSlidesConsumed();
+    }
+    if (slidesConsumed == null) {
       return;
     }
     if (beforeChange == null) {
-      updateParentSlides(parent, parent.getSlides() - childPiece.getSlidesConsumed(), changeUser);
+      updateParentSlides(parent, parent.getSlides() - slidesConsumed, changeUser);
```
Comment (callunity) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java:348]: I tried to test out consuming an absurd number of slides for stock 398048. I expected it to error or update the slides on parent 394809, but that slide parent was unchanged. Am I misunderstanding how the relationships should work?

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/SampleStockDto.java (hunk 1 above)
Comment (djcooke) [reply miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/SampleStockDto.java:13]: Yes. It's already on `SampleDto` and the override here was doing nothing

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java (hunk 2 above)
Comment (djcooke) [reply miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java:348]: That stock is not parented directly to a slide

Comment (callunity) [reply miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java:348]: Thanks!

**Review by callunity — APPROVED** [review]

**Review by avarsava — APPROVED** [review]

**Pull request merged**